src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -3521,7 +3521,7 @@ void MacroAssembler::vblendvps(XMMRegister dst, XMMRegister src1, XMMRegister sr
   // WARN: Allow dst == (src1|src2), mask == scratch
   bool blend_emulation = EnableX86ECoreOpts && UseAVX > 1;
   bool scratch_available = scratch != xnoreg && scratch != src1 && scratch != src2 && scratch != dst;
-  bool dst_available = dst != mask && (dst != src1 || dst != src2);
+  bool dst_available = (dst != mask || compute_mask) && (dst != src1 || dst != src2);
```

Comment:
I had thought of using `scratch_available` instead of `compute_mask` (i.e. `...dst_available = (dst != mask || scratch_available)...` but I thought it would make more sense to use `compute_mask`. i.e. dst is available to modify on 3531 because of the branch on `compute_mask` on 3526 

Also I prefer the shape of condition on 3525; three "simpler" independent boolean conditions instead of two longer ones. (i.e. Making `dst_available` depend on directly on `scratch_available` would make the it two "complex" conditions instead of three "simpler" conditions.) Though simplicity of the condition is debatable, I made it as readable as I could.